### PR TITLE
Added support of Nginx. Fix admin static files

### PR DESCRIPTION
**Pull request by matacoder**
Now we have 3 containers and working static files.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,8 +7,8 @@ This project provides a useful starter dough that you can use to bake a deliciou
 
 - A basic `Pipfile` with some useful packages and scripts.
 - A `docker-compose` file that allows hot-seat reloading and includes a PostgreSQL database that will automatically hook up to your Django app.
-- `gunicorn` for launching your webapp with several workers and threads. Makes it more scalable, basically.
-- `whitenoise` which is super convenient for serving static files straight out of your webapp. Greatly simplifies things.
+- `gunicorn` for launching your webapp with several workers and threads. Makes it more scalable, basically. 
+- 'nginx' for serving your static files. Run 'python manage.py collectstatic --noinput' in your project's directory
```
Comment (lemonsaurus) [inline README.md]: For consistency, let's use backticks here. 

```suggestion
- `nginx` for serving your static files. Run `python manage.py collectstatic --noinput` in your project's directory
```

File: .gitignore [hunk 2]
```diff
@@ -60,6 +60,8 @@ coverage.xml
 local_settings.py
 db.sqlite3
 db.sqlite3-journal
+static/
+media/
```
Comment (lemonsaurus) [inline .gitignore]: Would users have to create these folders manually, or would it happen when the docker-compose runs? Would docker-compose create real folders or just symlinks? 

One thing we could do is just add the _contents_ of these folders to the `.gitignore`, and add the folders themselves to the repo. Git will not allow us to add empty folders, but there's a convention to put an empty file named `.gitkeep` in a folder you want to keep, and make sure the `.gitignore` ignores everything except this file.

Would that make the user experience better?

File: README.md (hunk 1 above)
Comment (lemonsaurus) [reply README.md]: Also - I think you should add more information about the collectstatic command. What does it do? Why are we asking the user to run it?

File: docker-compose.yaml [hunk 3]
```diff
@@ -20,11 +20,21 @@ services:
...
+    container_name: django-starter-nginx
+    image: nginx:1.19.0-alpine
+    ports:
+      - "8020:80"
+    volumes:
+      - ./static:/code/static
+      - ./media:/code/media
+      - ./nginx:/etc/nginx/conf.d/
+    depends_on:
+      - hub
```
Comment (lemonsaurus) [inline docker-compose.yaml]: Can we change the name `hub` in this PR as well? I don't think we need it to be called hub, that was just a leftover from the project I adapted it from. It makes more sense to me to call it something like `django-project` for this generic starter, and encourage users to change it when they use this project.

**Review by lemonsaurus — CHANGES_REQUESTED** [review]
Looks pretty good, I have a few suggestions for changes.

File: .gitignore (hunk 2 above)
Comment (matacoder) [reply .gitignore]: That is a good idea. Added these folders.

File: docker-compose.yaml (hunk 3 above)
Comment (matacoder) [reply docker-compose.yaml]: Done

File: README.md (hunk 1 above)
Comment (matacoder) [reply README.md]: Done

File: .gitignore (hunk 2 above)
Comment (lemonsaurus) [reply .gitignore]: but shouldn't these two lines be removed now that we have the contents further up?

**Review by lemonsaurus — CHANGES_REQUESTED** [review]
Looks better, but we should probably remove the media and static folders from the .gitignore now that we're ignoring the contents, and I still don't see sufficient documentation about the purpose of collectstatic in this context.

File: .gitignore (hunk 2 above)
Comment (matacoder) [reply .gitignore]: There are two ways of doing this. You suggest using `.gitkeep` so we need to add "ignore folder" rule and "do not ignore file" rule at the same time. The second approach is to use no `.gitkeep` file but separate `.gitignore` file in every designated folder with content: 

> \*
> */
> !.gitignore

I chose option with 4 lines of code. The second is 6 lines long.

Comment (matacoder) [reply .gitignore]: https://stackoverflow.com/questions/4250063/how-to-gitignore-all-files-folder-in-a-folder-but-not-the-folder-itself

Comment (lemonsaurus) [reply .gitignore]: No, you're still using 6 lines of code. 

```
media/*
!media/.gitkeep
static/*
!static/.gitkeep
```

This is the solution, right? but why are we additionally also doing this?

```
static/
media/
```

File: .gitignore [hunk 4]
```diff
@@ -6,6 +6,12 @@ __pycache__/
 # C extensions
 *.so
 
+# Static and Media contents
+media/*
+!media/.gitkeep
+static/*
+!static/.gitkeep
```
Comment (lemonsaurus) [inline .gitignore]: similar things look better together.

```suggestion
media/*
static/*
!media/.gitkeep
!static/.gitkeep
```

**Review by lemonsaurus — CHANGES_REQUESTED** [review]
99% there.

File: .gitignore (hunk 2 above)
Comment (matacoder) [reply .gitignore]: I've just missed them, sorry for that. Corrected

**Review by lemonsaurus — APPROVED** [review]

**Pull request merged**